### DISCUSSION TITLE
best way to add an ip address to a vlan of a switch?

### DISCUSSION THREAD

**Discussion by LeDistordu** [netbox-community/netbox#13483] [Q&A]
Hi :) Currently you can only add an IP address to an interface. Is it possible to add a vlan to a switch, to add an IP to it? A bypass?

**Comment by jeremystretch** [comment]
> Currently you can only add an IP address to an interface.

This is a reflection of a real-world limitation: An interface **must** exist to effect IP routing. Depending on the platform, this may not be readily apparent, but the assigned of an IP address to "a VLAN" _always_ requires the designation of an interface. This is often referred to as a Switched Virtual Interface (SVI) mapped to the L2 domain internally on the device. You can create these in NetBox as virtual interfaces.

**Comment by candlerb** [comment]
> Is it possible to add a vlan to a switch, to add an IP to it?

Yes. #13026

**Comment by LeDistordu** [comment]
Thank you it's ok for me :)
